### PR TITLE
[incubator/kafka] Add support for Static Ips for Kafka External Load Balancer

### PR DESCRIPTION
**Pull request by akshayzendesk**
Signed-off-by: Akshay Agrawal <aagrawal@zendesk.com>

#### What this PR does / why we need it:
This adds support for static IP's for Kafka External if type is "LoadBalancer". This can be used in cloud providers.

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer:
We are not making much changes here. Just adding support for static IP's in kafka external service.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/kafka/templates/service-brokers-external.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@
     {{- $externalListenerPort := add $root.Values.external.firstListenerPort $i }}
     {{- $responsiblePod := printf "%s-%d" (printf "%s" $fullName) $i }}
     {{- $distinctPrefix := printf "%s-%d" $dnsPrefix $i }}
+    {{- $loadBalancerIP := index $root.Values.external.loadBalancerIP $i }}
```
Comment (agolomoodysaada) [inline incubator/kafka/templates/service-brokers-external.yaml:11]: If not loadbalancer ips are set, it throws

```
Error: render error in "kafka/charts/kafka/templates/service-brokers-external.yaml": template: kafka/charts/kafka/templates/service-brokers-external.yaml:11:27: executing "kafka/charts/kafka/templates/service-brokers-external.yaml" at <index $root.Values.e...>: error calling index: index out of range: 0
```

Can we make the static ip optional ?

**Review by agolomoodysaada — COMMENTED** [review]
There is a breaking change here

**Comment by akshayzendesk** [comment]
@agolomoodysaada I ll get to it ASAP.

**Comment by casidiablo** [comment]
@akshayzendesk can you please fix this?

**Comment by kaarolch** [comment]
@akshayzendesk   this PR generate: 
```
Error: render error in "kafka/templates/service-brokers-external.yaml": template: kafka/templates/service-brokers-external.yaml:11:27: executing "kafka/templates/service-brokers-external.yaml" at <index $root.Values.e...>: error calling index: index out of range: 0
```
When default `loadBanalncerIP: []` and you use nodePort.

File: incubator/kafka/templates/service-brokers-external.yaml (hunk 1 above)
Comment (kaarolch) [reply incubator/kafka/templates/service-brokers-external.yaml:11]: Added PR #10816

**Pull request merged**